### ISSUE TITLE
Account Creation

### ISSUE THREAD

**Issue by timowens9** [chingu-voyages/bears-project-3#4]
As a new potential user, I want to create and register a new account so I can log in and use the application.

Considerations: User name and password fields, Two buttons, one for sign in and one for register

Testing: Input form validation
check db if user exists

**Comment by adilasif** [comment]
Would we like to have two seperate forms or use the same form with two buttons appended? I'm envisioning a registration form that asks for information including name, location, date of birth, etc. and a sign-in form as a separate form that asks for email or username and password. Alternately, we could just ask for email and password and share forms, and users can then fill out their name, location, etc. on their profile page, perhaps.

Do we want these forms to be overlays, without redirecting the user to another page, or do we want discrete web pages?

**Comment by adilasif** [comment]
Good idea on setting up oauth signing with social media, we should definitely set that up if we can, along with some form of 2 factor authentication - passwords are essentially obsolete these days, after all.

**Comment by timowens9** [comment]
how about something like this
auth
seems pretty straight forward

**Comment by adilasif** [comment]
Check out https://auth0.com - they are an api that handles all auth for us, so we don't have to worry about implementing our own encryption all over the place. Works with all stacks, can be used front end or back end, has Postman integration, and is easily extensible for our needs, so we can add 2fa at any time later on after setting up a super basic signin handler.

**Comment by timowens9** [comment]
This looks good to me.

**Comment by NariRoh** [comment]
> Check out https://auth0.com - they are an api that handles all auth for us, so we don't have to worry about implementing our own encryption all over the place. Works with all stacks, can be used front end or back end, has Postman integration, and is easily extensible for our needs, so we can add 2fa at any time later on after setting up a super basic signin handler.
> 
> On the other hand, if you want the experience of rolling our own encrypted data pipeline, I'm open to that too!

I'm down!

**Comment by iamlasse** [comment]
An Account Profile could have a listing of upcoming events they've "joined" and some details about these events, like other attendees, date, location, price etc.

Should we have some settings on the profile, for example, notifications and which to allow/reject?

**Comment by iamlasse** [comment]
A meetup profile for inspiration/reference

![Screen Shot 2018-12-08 at 2.34.50 PM.png](https://images.zenhubusercontent.com/5c0a8320c57eec537ab64a39/2bac4644-e156-4799-a68c-1152ec1f61a4)
